CI Failure when no tests are collected

**Describe the bug**
E2E tests fail on `main` because we currently have no "smoke" tests for E2E.

**Expected behavior**
The GitHub action should ignore failures due to 0 tests being collected. #469 contained a check for this but it does not work.

**Errors**
```
============================= test session starts ==============================
platform linux -- Python 3.10.19, pytest-8.2.2, pluggy-1.6.0 -- /home/runner/work/guidellm/guidellm/.tox/test-e2e/bin/python
cachedir: .tox/test-e2e/.pytest_cache
rootdir: /home/runner/work/guidellm/guidellm
configfile: pyproject.toml
plugins: anyio-4.11.0, respx-0.22.0, timeout-2.4.0, Faker-37.11.0, cov-5.0.0, mock-3.14.1, rerunfailures-14.0, asyncio-1.1.1
asyncio: mode=strict, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collecting ... collected 4 items / 4 deselected / 0 selected

============================ 4 deselected in 0.32s =============================
test-e2e: exit 5 (2.30 seconds) /home/runner/work/guidellm/guidellm> python -m pytest tests/e2e -m smoke pid=3129
  test-e2e: FAIL code 5 (34.88=setup[32.59]+cmd[2.30] seconds)
  evaluation failed :( (34.92 seconds)
Error: Process completed with exit code 5.
```

**Additional context**
Non-working bit here: https://github.com/vllm-project/guidellm/blob/765c875a7378d916fe22accd6b66371062a43d54/.github/actions/run-tox/action.yml#L27-L28

--- a/.github/actions/run-tox/action.yml
+++ b/.github/actions/run-tox/action.yml
@@ -23,7 +23,7 @@ runs:
       shell: bash
     - name: Run tox
       run: |
-        tox run -e "${{ inputs.tox-env }}" -- ${{ inputs.tox-args }}
+        tox run -e "${{ inputs.tox-env }}" -- ${{ inputs.tox-args }} || ret=$?
         # errno 5 means all tests were filtered out, ignore
-        ret=$? && [[ $ret -eq 5 ]] && exit 0; exit $ret
+        [[ "${ret:=0}" -eq 5 ]] && exit 0; exit $ret
       shell: bash

